grouper/setup.py

```diff
@@ -51,15 +53,13 @@ def build_arg_parser(description):
 
 def setup_logging(args, log_format):
     # type: (Namespace, str) -> None
-
-    # `logging` levels are integer multiples of 10. so each verbose/quiet level
-    # is multiplied by 10
-    verbose = args.verbose * 10
-    quiet = args.quiet * 10
-    log_level = logging.getLogger().level - verbose + quiet
+    log_level = get_loglevel(args)
 
     sa_log = logging.getLogger("sqlalchemy.engine.base.Engine")
 
     logging.basicConfig(level=log_level, format=log_format)
     if log_level < 0:
         sa_log.setLevel(logging.INFO)
+
+    for handler in logging.root.handlers:
+        handler.setFormatter(RedactingFormatter(log_format))
```

Comment:
If you wanted to retain the original formatters, you could use logic similar to [this blog post](https://relaxdiego.com/2014/07/logging-in-python.html) that sets up the formatter with all the other formatters, and applies those first before applying the redactions. You would also need to adjust the logic to `init` the class + the `format` function to take into account what formatters the `RedactingFormatter` was initialized with.

Reply:
I've looked into this quite a bit, and the typing system makes it very complicated. While the native python type system will gladly work with the `get_attr` approach to "proxying" the objects, mypy refuses it (with good reason). I'm going to stick with an approach similar to what I have here, since it works with the more common object-oriented type system that mypy presents at the surface.